### PR TITLE
Add xterm@2.3.2 w/ git auto-update

### PR DESCRIPTION
**Pull request by jagadeeshv16**
Pull request for issue: #
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/sourcelair/xterm.js
 * Official website (optional, not the repository): https://xtermjs.org/
 * NPM package url (optional): https://www.npmjs.com/package/xterm
 * License and its reference: MIT(https://github.com/sourcelair/xterm.js/blob/master/LICENSE)
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 45
   - Count of stars: 1231
   - Count of forks: 150
 * NPM download stats (optional):
   - Downloads in the last day: 145
   - Downloads in the last week: 2683
   - Downloads in the last month: 8830

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/xterm/package.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    "type": "git",
+    "url": "git+https://github.com/sourcelair/xterm.js"
+  },
+  "keywords": [
+    "xterm",
+    "terminal",
+    "browser-tty"
+  ],
+  "author": "SourceLair",
+  "license": "MIT"
```
Comment (hare1039) [inline ajax/libs/xterm/package.json]: @pvnr0082t Do we require to have auto-update for every PR?

File: ajax/libs/xterm/package.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "xterm",
+  "filename": "xterm.js",
```
Comment (hare1039) [inline ajax/libs/xterm/package.json]: We should include/generate `.min` files, so please change filename to `xterm.min.js`

Our recommend minify tool is this one: https://github.com/PeterDaveHello/web-minify-helper

**Comment by pvnr0082t** [comment]
@hare1039 You can directly help PR author or ask him to fill the information in the initial PR comment. :+1:

File: ajax/libs/xterm/package.json (hunk 1 above)
Comment (pvnr0082t) [inline ajax/libs/xterm/package.json]: Sure. If there are git tags or npm releases, we need to set auto-update config for the future updating.

**Comment by jagadeeshv16** [comment]
@pvnr0082t, @hare1039 Apart from adding .min files and reference them in package.json, are there any other required changes?

**Comment by hare1039** [comment]
Would you add the auto-update config at the buttom?
```
  "autoupdate": {
    "source": "git",
    "target": "https://github.com/sourcelair/xterm.js.git",
    "fileMap": [
      {
        "basePath": "dist",
        "files": [
          "**/*"
        ]
      }
    ]
  }
```

**Comment by hare1039** [comment]
Also please generate the minified files.
If it is not convenient for you to generate minified files, please tell us.

**Comment by jagadeeshv16** [comment]
@hare1039 i created minfiles and added auto update code as you requested.

**Comment by jagadeeshv16** [comment]
@sufuf3 updated the package,json with homepage and file poiniting to xterm.min.js and updated the PR with the stats requested

File: ajax/libs/xterm/package.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  "keywords":[
+    "xterm",
+    "terminal",
+    "browser-tty"
+  ],
+  "author":"SourceLair",
+  "license":"MIT",
+  "autoupdate":{
+    "source":"git",
+    "target":"https://github.com/sourcelair/xterm.js.git",
```
Comment (pvnr0082t) [inline ajax/libs/xterm/package.json]: @jagadeeshv16 Please follow the target format in [autoupdate.md](https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md). Let's use `git://github.com/sourcelair/xterm.js.git`.

Comment (jagadeeshv16) [inline ajax/libs/xterm/package.json]: @pvnr0082t updated the target information

**Comment by jagadeeshv16** [comment]
@pvnr0082t  changed url in autocomplete section of package.json.

**Comment by PeterDaveHello** [comment]
@sufuf3 please help squash/fixup the commits and rebase its branch upon the latest master, thanks.

**Comment by jagadeeshv16** [comment]
@sufuf3 any update?

**Comment by sufuf3** [comment]
@cdnjs/intern3 please help review this PR, thanks.

**Comment by sufuf3** [comment]
I'm really sorry for pending too long, I can't minify the `xterm.js` of v0.4.
So I'll manually add old versions after this PR merge.
I help rebase master, @cdnjs/intern3 please review this PR.

Thank you.

File: ajax/libs/xterm/package.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  "description": "Full xterm terminal, in your browser",
+  "homepage": "https://xtermjs.org/",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/sourcelair/xterm.js"
+  },
+  "keywords": [
+    "xterm",
+    "terminal",
+    "browser-tty"
```
Comment (extend1994) [inline ajax/libs/xterm/package.json]: Is it a type error here?
If yes, please modify it and also add some other keywords to let users be able to find the library in the first page of the searching result in a search engine like Google. E.g `JavaScript` and `browser`.

Comment (PeterDaveHello) [inline ajax/libs/xterm/package.json]: We don't need to be that strict on this part for non-members.

**Comment by PeterDaveHello** [comment]
Ping @hare1039 @pvnr0082t @clairetsai818

File: ajax/libs/xterm/package.json (hunk 4 above)
Comment (extend1994) [inline ajax/libs/xterm/package.json]: Ok :+1:

**Comment by PeterDaveHello** [comment]
What's the current progress here please? @cdnjs/intern3

**Comment by sufuf3** [comment]
Ping @hare1039 @extend1994

**Pull request merged**